packages/apm/src/span.ts

```diff
@@ -21,8 +21,9 @@ class SpanRecorder {
   private _openSpanCount: number = 0;
   public finishedSpans: Span[] = [];
 
-  public constructor(maxlen: number) {
-    this._maxlen = maxlen;
+  public constructor(maxlen?: number) {
+    // tslint:disable-next-line: strict-type-predicates
+    this._maxlen = typeof maxlen !== 'number' ? 1000 : maxlen;
```

Comment:
In the version of TS we're using, could we do this?

```suggestion
  public constructor(maxlen = 1000) {
    this._maxlen = maxlen;
```

https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

> In TypeScript, we can also set a value that **a parameter will be assigned if the user does not provide one**, **or if the user passes _undefined_** in its place. These are called default-initialized parameters.

Reply:
So the code before was good ^^